src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -199,14 +204,26 @@ export class SavedSearchEmbeddable
           executionContext,
         })
       );
-      this.updateOutput({ loading: false, error: undefined });
+
+      this.updateOutput({
+        ...this.getOutput(),
+        // TODO: rendering happens later - separate the rendered and loading events
+        rendered: true,
```

Comment:
wouldn't it make more sense to not change `rendered` here, but to set it to true here, after  ReactDOM.render? 

https://github.com/elastic/kibana/blob/dcb10bd6150342d0d0e8ceee15ab501d11b7da18/src/plugins/discover/public/embeddable/saved_search_embeddable.tsx#L406

Reply:
We synced about this and @kertal will check if we can actually use `renderReactComponent`

Reply:
Fix is on the way: https://github.com/elastic/kibana/pull/134891

Reply:
Now reporting rendered from `renderReactComponent`!